### PR TITLE
Create website for testing Token generation #16

### PR DESCRIPTION
**Pull request by munyanezaarmel**
### Brief summary of the change made
 - added index.html file  which includes everything for OpenPaygo token website

Closes: #16 

### Are there any other side effects of this change that we should be aware of?
none

### Describe how you tested your changes?
- by referring to the unit tests.  token matches the format of the token generated in the browser

image

### Pull Request checklist

Please confirm you have completed any of the necessary steps below.

- [x] Meaningful Pull Request title and description
- [x] Changes tested as described above
- [x] Added appropriate documentation for the change.
- [x] Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

**Comment by munyanezaarmel** [comment]
@dmohns can i get your review

**Comment by dmohns** [comment]
Nice, thanks @munyanezaarmel the website look good.

However, you are essentially hardcoding the entire repositories code into the HTML file, which is not exactly what we want.

Could me make it more like this:

- Use a JavaScript website framework of your choice
- Install the OpenPAYGO-js using `npm`

Build the website using `npm` production build.

**Comment by munyanezaarmel** [comment]
Hello @dmohns,

Thank you for your review. I've decided to proceed with React/Next.js for my project. However, I'm encountering an issue with package accessibility. Despite successful installation, I'm unable to access the package in my code.
A significant concern is the absence of an index.js file that clearly shows which functions are available to be called. This omission is causing confusion, as it's unclear which functions are intended for use and which are not.
Could you please provide documentation on how to properly use this package? Specifically, it would be helpful to have:

- A clear list of exported functions and their purposes
- Examples of how to import and use these functions
- Any configuration or setup required

For reference, I'm looking for something similar to the documentation style found on the npm website for packages like 'decoders' (https://www.npmjs.com/package/decoders).
image

File: src/index.html [hunk 1]
```diff
@@ -0,0 +1,345 @@
+<!DOCTYPE html>
```
Comment (dmohns) [inline src/index.html:1]: This file can be deleted now, right?

File: openpaygo-token-generator/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+This is a [Next.js](https://nextjs.org) project bootstrapped with [`create-next-app`](https://nextjs.org/docs/app/api-reference/cli/create-next-app).
```
Comment (dmohns) [inline openpaygo-token-generator/README.md:1]: Could you update and clean up the `README`? Keep the build instructions, but explain here what this demo is supposed to do and remove the `nextjs` boilerplate.

File: openpaygo-token-generator/src/openpaygo.d.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+
+declare module 'openpaygo' {
+    export class OpenPAYGOTokenEncoder {
+        constructor();
+        generateToken(params: {
```
Comment (dmohns) [inline openpaygo-token-generator/src/openpaygo.d.ts:5]: Not a JavaScript/TypeScript expert. Is this file really required?

**Review by dmohns — COMMENTED** [review]
Few comments in the code.

Could you move the entire website generation code into a folder called `website` rather than `openpaygo-token-generator` ?

File: openpaygo-token-generator/src/openpaygo.d.ts (hunk 3 above)
Comment (munyanezaarmel) [reply openpaygo-token-generator/src/openpaygo.d.ts]: Yes, the openpaygo.d.ts file is required because we are using the openpaygo module in a TypeScript project and the module does not have its own type definitions, Since we are using the openpaygo module and encountering type-related errors, having the openpaygo.d.ts file is necessary to ensure that  TypeScript code is type-safe and to avoid runtime errors.  and also If we find that the types we defined in the declaration file are insufficient, we can always expand it as needed.

Comment (dmohns) [reply openpaygo-token-generator/src/openpaygo.d.ts]: Ok, understood. But this sounds like a feature request for the JS OpenPAYGO library, right? I.e. providing Type definitions such that the project can easily be integrated into TypeScript codebases.

@beesaferoot does this make sense to you?

@munyanezaarmel could you create a seperate Github issue for this feature request? We can leave your workaround here for now.

Comment (beesaferoot) [reply openpaygo-token-generator/src/openpaygo.d.ts]: @dmohns Yes, I think we can go with this pending when we include type definitions for the JS library.

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+name: Deploy to GitHub Pages
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```
Comment (dmohns) [inline .github/workflows/deploy.yml:7]: Please don't run deploy jobs on Pull Requests with out filter !
```suggestion
on:
  # Runs on pushes targeting the default branch
  push:
    branches: [main]
```

Comment (munyanezaarmel) [reply .github/workflows/deploy.yml:7]: @dmohns please check again

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
```
Comment (dmohns) [inline .github/workflows/deploy.yml:7]: Please remove the entire `pull_request:` line here 😉

File: .github/workflows/deploy.yml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+      - name: Install dependencies
+        run: npm install
+
+      - name: Build and Export
+        run: |
+          npm run build
+          npm run export
+
+      - name: Deploy to GitHub Pages
+        uses: peaceiris/actions-gh-pages@v3
```
Comment (dmohns) [inline .github/workflows/deploy.yml:57]: Please use the official Github Action to deploy the pages, i.e. `actions/deploy-pages@v2`

Comment (munyanezaarmel) [reply .github/workflows/deploy.yml]: @dmohns can you please check again

**Review by dmohns — DISMISSED** [review]

File: .github/workflows/deploy.yml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+      - name: Setup Pages
+        uses: actions/configure-pages@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "18"
+          cache: "npm"
```
Comment (dmohns) [inline .github/workflows/deploy.yml:42]:
```suggestion
          cache: npm
```

**Review by dmohns — APPROVED** [review]
